src/components/core/compute.ts

```diff
@@ -53,3 +61,143 @@ export class GetEnvironmentsHandler extends Handler {
     }
   }
 }
+
+export class InitializeComputeHandler extends Handler {
+  validateTimestamp(value: number) {
+    // in miliseconds
+    const timestampNow = new Date().getTime() / 1000
+    const validUntil = new Date(value).getTime() / 1000
+
+    return validUntil > timestampNow
+  }
+
+  checksC2DEnv(computeEnv: string, c2dEnvsWithHash: any[]): boolean {
+    for (const c of c2dEnvsWithHash) {
+      if (c.id === computeEnv) {
+        return true
+      }
+    }
+    return false
+  }
+
+  getServiceById(ddo: DDO, serviceId: string) {
+    try {
+      return ddo.services.filter((service) => service.id === serviceId)[0]
+    } catch (err) {
+      CORE_LOGGER.logMessage(`Service was not found: ${err}`)
```

Comment:
It would be better to log an error rather than just a message: 

```suggestion
      CORE_LOGGER.error(`Service was not found: ${err}`)
```